addons/calendar_sms/tests/test_calendar_sms.py

```diff
@@ -31,3 +77,30 @@ def test_attendees_with_number(self):
             'partner_ids': [(6, 0, [self.partner_phone.id, self.partner_no_phone.id])],
         })._sms_get_default_partners()
         self.assertEqual(len(attendees), 1, "There should be only one partner retrieved")
+
+    def test_send_reminder_match_both_events(self):
+        """
+        Test that only the necessary SMS messages are sent,
+         with each SMS template correctly matching its corresponding alarm duration and the event.
+        """
+        with self.mockSMSGateway():
+            lastcall = fields.Datetime.now() - timedelta(hours=1)
+            self.env['calendar.alarm_manager'].with_context(lastcall=lastcall)._send_reminder()
+
+            mail_message_24h = self.env['mail.message'].search([
+                ('model', '=', 'calendar.event'),
+                ('res_id', '=', self.event_24h.id),
+                ('message_type', '=', 'sms'),
+            ])
+
+            mail_message_1h = self.env['mail.message'].search([
+                ('model', '=', 'calendar.event'),
+                ('res_id', '=', self.event_1h.id),
+                ('message_type', '=', 'sms'),
+            ])
+
+            self.assertEqual(html.fromstring(str(mail_message_24h.body)).text_content(),self.sms_template_24h.body)
+            self.assertEqual(html.fromstring(str(mail_message_1h.body)).text_content(),self.sms_template_1h.body)
+            self.assertEqual(len(self._sms), 2)
```

Comment:
:thinking: Can't you use the assertSMS helper? I think that'd be the cleanest, and would avoid calling lxml
Would just need to make sure both events have different partners/numbers